### PR TITLE
media-gfx/img2pdf: Version bump to 0.5.1

### PR DESCRIPTION
**Pull request by gringus**
Version bump to `0.5.1` for `media-gfx/img2pdf`
Dropped `ppc64` arch due to dependency on `media-libs/icc-profiles-openicc` which is not supported on `ppc64`

---

Please check all the boxes that apply:

- [*] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [*] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [*] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [*] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{9..11} )
```
Comment (a17r) [inline media-gfx/img2pdf/img2pdf-0.5.1.ebuild:7]: Could you also enable py3.12 here please (and run tests)?

**Review by a17r — CHANGES_REQUESTED** [review]

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild (hunk 1 above)
Comment (a17r) [reply media-gfx/img2pdf/img2pdf-0.5.1.ebuild:7]: (see also https://bugs.gentoo.org/929621)

Comment (gringus) [reply media-gfx/img2pdf/img2pdf-0.5.1.ebuild]: Enabled 3.12, tested (3.11 & 3.12 were successful) and pushed.

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	dev-python/pikepdf[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	media-libs/icc-profiles-openicc
+	gui? ( $(python_gen_impl_dep tk) )
+"
+
+distutils_enable_tests pytest
+
+EPYTEST_DESELECT=(
+	# Failing cmyk8 tests
```
Comment (mgorny) [inline media-gfx/img2pdf/img2pdf-0.5.1.ebuild:44]: That doesn't really tell us anything. Either tell us why they're failing, ideally link upstream bug report, or just remove the comment.

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{9..12} )
```
Comment (mgorny) [inline media-gfx/img2pdf/img2pdf-0.5.1.ebuild:7]: Remove 3.9, it's no longer used.

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+inherit distutils-r1 pypi
+
+DESCRIPTION="Losslessly convert raster images to PDF"
+HOMEPAGE="https://gitlab.mister-muffin.de/josch/img2pdf"
```
Comment (mgorny) [inline media-gfx/img2pdf/img2pdf-0.5.1.ebuild:12]:
```suggestion
HOMEPAGE="
	https://gitlab.mister-muffin.de/josch/img2pdf/
	https://pypi.org/project/img2pdf/
"
```

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	# Failing cmyk8 tests
+	src/img2pdf_test.py::test_miff_cmyk8
+)
+
+src_prepare() {
+	distutils-r1_python_prepare_all
+
+	# Remove gui executable if there's no demand/support for it.
+	if ! use gui; then
+		sed -i '/gui_scripts/d' setup.py || die
```
Comment (mgorny) [inline media-gfx/img2pdf/img2pdf-0.5.1.ebuild:53]:
```suggestion
		sed -i -e '/gui_scripts/d' setup.py || die
```

I'm not sure why we use `-e` everywhere but we generally prefer consistency here.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild (hunk 3 above)
Comment (gringus) [reply media-gfx/img2pdf/img2pdf-0.5.1.ebuild]: Dropped

File: media-gfx/img2pdf/img2pdf-0.5.1.ebuild (hunk 2 above)
Comment (gringus) [reply media-gfx/img2pdf/img2pdf-0.5.1.ebuild]: Linked bug report

**Review by mgorny — COMMENTED** [review]
I'm getting more test failures:

```pytb
FAILED src/img2pdf_test.py::test_miff_cmyk16[internal] - assert 6.35075 > 50
FAILED src/img2pdf_test.py::test_miff_cmyk16[pikepdf] - assert 6.35075 > 50
```

Do these two pass for you?

**Comment by gringus** [comment]
Yes, all pass.. interesting...

**Comment by a17r** [comment]
All passed here too: `341 passed, 8 skipped, 2 deselected in 41.02s`

**Comment by mgorny** [comment]
Do these tests run at all for you? I don't have a single skip, so I'm guessing they could be skipped for you because of missing deps:

```
2 failed, 347 passed, 2 deselected in 116.59s (0:01:56)
```

Logs for failures:
```pytb
_____________________________________________________ test_miff_cmyk16[internal] ______________________________________________________

tmp_path_factory = TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0x7f4909c630b0>, _basetemp=PosixP.../tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0')
, _retention_count=0, _retention_policy='failed')
miff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk160/in.miff')
tiff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
miff_cmyk16_pdf = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf0/out.pdf')

    @pytest.mark.skipif(
        sys.platform in ["win32"],
        reason="test utilities not available on Windows and MacOS",
    )
    def test_miff_cmyk16(
        tmp_path_factory, miff_cmyk16_img, tiff_cmyk16_img, miff_cmyk16_pdf
    ):
        tmpdir = tmp_path_factory.mktemp("miff_cmyk16")
>       compare_ghostscript(
            tmpdir, tiff_cmyk16_img, miff_cmyk16_pdf, gsdevice="tiff32nc", exact=False
        )

miff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk160/in.miff')
miff_cmyk16_pdf = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf0/out.pdf')
tiff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
tmp_path_factory = TempPathFactory(_given_basetemp=None,
                _trace=<pluggy._tracing.TagTracerSub object at 0x7f4909c630b0>,
                _basetemp=PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0'),
                _retention_count=0,
                _retention_policy='failed')
tmpdir     = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk161')

src/img2pdf_test.py:6422: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/img2pdf_test.py:418: in compare_ghostscript
    compare(str(img), str(tmpdir / "gs-1.") + ext, exact, icc, False)
        exact      = False
        ext        = 'tiff'
        gsdevice   = 'tiff32nc'
        icc        = False
        img        = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
        pdf        = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf0/out.pdf')
        tmpdir     = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk161')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

im1 = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff'
im2 = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk161/gs-1.tiff', exact = False, icc = False
cmyk = False

    def compare(im1, im2, exact, icc, cmyk):
        if exact:
            if cmyk and not HAVE_EXACT_CMYK8:
                raise Exception("cmyk cannot be exact before ImageMagick 7.1.0-48")
            elif icc:
                raise Exception("icc cannot be exact")
            else:
                subprocess.check_call(
                    COMPARE
                    + [
                        "-metric",
                        "AE",
                        "-alpha",
                        "off",
                        im1,
                        im2,
                        "null:",
                    ]
                )
        else:
            iccargs = []
            if icc:
                if ICC_PROFILE is None:
                    pytest.skip("Could not locate an ICC profile")
                iccargs = ["-profile", ICC_PROFILE]
            psnr = subprocess.run(
                COMPARE
                + iccargs
                + [
                    "-metric",
                    "PSNR",
                    im1,
                    im2,
                    "null:",
                ],
                check=False,
                stderr=subprocess.PIPE,
            ).stderr
            assert psnr != b"0"
            assert psnr != b"0 (0)"
            assert psnr_re.fullmatch(psnr) is not None, psnr
            psnr = psnr_re.fullmatch(psnr).group(1)
            psnr = float(psnr)
            assert psnr != 0  # or otherwise we would use the exact variant
>           assert psnr > 50
E           assert 6.35075 > 50
cmyk       = False
exact      = False
icc        = False
iccargs    = []
im1        = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff'
im2        = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk161/gs-1.tiff'
psnr       = 6.35075

src/img2pdf_test.py:396: AssertionError
______________________________________________________ test_miff_cmyk16[pikepdf] ______________________________________________________

tmp_path_factory = TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0x7f4909c630b0>, _basetemp=PosixP.../tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0')
, _retention_count=0, _retention_policy='failed')
miff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk160/in.miff')
tiff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
miff_cmyk16_pdf = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf1/out.pdf')

    @pytest.mark.skipif(
        sys.platform in ["win32"],
        reason="test utilities not available on Windows and MacOS",
    )
    def test_miff_cmyk16(
        tmp_path_factory, miff_cmyk16_img, tiff_cmyk16_img, miff_cmyk16_pdf
    ):
        tmpdir = tmp_path_factory.mktemp("miff_cmyk16")
>       compare_ghostscript(
            tmpdir, tiff_cmyk16_img, miff_cmyk16_pdf, gsdevice="tiff32nc", exact=False
        )

miff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk160/in.miff')
miff_cmyk16_pdf = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf1/out.pdf')
tiff_cmyk16_img = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
tmp_path_factory = TempPathFactory(_given_basetemp=None,
                _trace=<pluggy._tracing.TagTracerSub object at 0x7f4909c630b0>,
                _basetemp=PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0'),
                _retention_count=0,
                _retention_policy='failed')
tmpdir     = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk162')

src/img2pdf_test.py:6422: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/img2pdf_test.py:418: in compare_ghostscript
    compare(str(img), str(tmpdir / "gs-1.") + ext, exact, icc, False)
        exact      = False
        ext        = 'tiff'
        gsdevice   = 'tiff32nc'
        icc        = False
        img        = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff')
        pdf        = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk16_pdf1/out.pdf')
        tmpdir     = PosixPath('/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk162')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

im1 = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff'
im2 = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk162/gs-1.tiff', exact = False, icc = False
cmyk = False

    def compare(im1, im2, exact, icc, cmyk):
        if exact:
            if cmyk and not HAVE_EXACT_CMYK8:
                raise Exception("cmyk cannot be exact before ImageMagick 7.1.0-48")
            elif icc:
                raise Exception("icc cannot be exact")
            else:
                subprocess.check_call(
                    COMPARE
                    + [
                        "-metric",
                        "AE",
                        "-alpha",
                        "off",
                        im1,
                        im2,
                        "null:",
                    ]
                )
        else:
            iccargs = []
            if icc:
                if ICC_PROFILE is None:
                    pytest.skip("Could not locate an ICC profile")
                iccargs = ["-profile", ICC_PROFILE]
            psnr = subprocess.run(
                COMPARE
                + iccargs
                + [
                    "-metric",
                    "PSNR",
                    im1,
                    im2,
                    "null:",
                ],
                check=False,
                stderr=subprocess.PIPE,
            ).stderr
            assert psnr != b"0"
            assert psnr != b"0 (0)"
            assert psnr_re.fullmatch(psnr) is not None, psnr
            psnr = psnr_re.fullmatch(psnr).group(1)
            psnr = float(psnr)
            assert psnr != 0  # or otherwise we would use the exact variant
>           assert psnr > 50
E           assert 6.35075 > 50

cmyk       = False
exact      = False
icc        = False
iccargs    = []
im1        = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/tiff_cmyk160/in.tiff'
im2        = '/tmp/portage/media-gfx/img2pdf-0.5.1/temp/pytest-of-portage/pytest-0/miff_cmyk162/gs-1.tiff'
psnr       = 6.35075

src/img2pdf_test.py:396: AssertionError
```
```

**Comment by gringus** [comment]
I get `349 passed, 2 deselected in 33.11s`

**Comment by gringus** [comment]
Pkease try the patch from https://gitlab.mister-muffin.de/josch/img2pdf/issues/178 and see if that helps.

**Comment by mgorny** [comment]
The NixOS patch? It's for cmyk8 cases only. The author even notes that cmyk16 cases still fail for them after applying it.

Perhaps it's ImageMagick flags? I've got 7.1.1.25, `USE="X bzip2 cxx djvu fftw fontconfig fpx hdri heif jbig jpeg jpeg2k jpegxl lcms lqr lzma openexr openmp pango perl png postscript raw svg tiff truetype webp wmf xml zlib"`.

**Comment by mgorny** [comment]
If I limit the flags to absolute minimum, the tests pass. I'm going to try different flag combinations to figure out which one causes the problem.

**Comment by mgorny** [comment]
Ok, it's `USE=hdri`. Let's just add these two tests to deselect (possibly with `if has_version 'media-gfx/imagemagick[hdri]`), no point in forcing it off for the sake of tests.

**Review by mgorny — APPROVED** [review]
Thanks. I'm going to add xdist back, because the tests are slow, then push and open rekeywording.

**Comment by mgorny** [comment]
Do you want to maintain it going forward, by the way?

**Pull request closed without merging**